### PR TITLE
Fix: Finder incorrectly identifies pages created [ED-1269]

### PR DESCRIPTION
**Pull request by iNewLegend**

### CODE REVIEW

File: core/documents-manager.php [hunk 1]
```diff
@@ -693,4 +693,15 @@ private function register_types() {
 			do_action( 'elementor/documents/register', $this );
 		}
 	}
+
+	public function create_new_post_url( $type ) {
```
Comment (matipojo) [inline core/documents-manager.php:697]:
```suggestion
	public function get_create_new_post_url( $type ) {
```

File: core/base/document.php [hunk 2]
```diff
@@ -1258,4 +1259,14 @@ protected function get_post_statuses() {
 	protected function get_have_a_look_url() {
 		return $this->get_permalink();
 	}
+
+	public function get_name() {
+		return static::get_type();
+	}
+
+	public static function get_create_url() {
+		$base_create_url = Utils::get_create_new_post_url( Source_Local::CPT );
```
Comment (matipojo) [inline core/base/document.php:1268]: Move from utils to the documents manager.

**Review by matipojo — CHANGES_REQUESTED** [review]

File: core/documents-manager.php (hunk 1 above)
Comment (iNewLegend) [reply core/documents-manager.php:697]: Handled.

File: core/base/document.php (hunk 2 above)
Comment (iNewLegend) [reply core/base/document.php]: Handled.

**Review by matipojo — CHANGES_REQUESTED** [review]

File: core/common/modules/finder/categories/create.php [hunk 3]
```diff
@@ -54,7 +56,8 @@ public function get_category_items( array $options = [] ) {
 			if ( Source_Local::CPT === $post_type ) {
 				$url = admin_url( Source_Local::ADMIN_MENU_SLUG . '#add_new' );
 			} else {
-				$url = Utils::get_create_new_post_url( $post_type );
+				$doc_type = Plugin::$instance->documents->get_document_type( $post_type );
```
[the remark this answers is not in the document]
Comment (iNewLegend) [reply core/common/modules/finder/categories/create.php]: Handled.

**Comment by shilo-ey** [comment]
ref: https://github.com/elementor/elementor/issues/12502

File: core/common/modules/finder/categories/create.php [hunk 4]
```diff
@@ -2,6 +2,8 @@
 namespace Elementor\Core\Common\Modules\Finder\Categories;
 
 use Elementor\Core\Common\Modules\Finder\Base_Category;
+use Elementor\Core\Documents_Manager;
```
Comment (matipojo) [inline core/common/modules/finder/categories/create.php:5]: Why?

**Review by matipojo — CHANGES_REQUESTED** [review]

File: core/common/modules/finder/categories/create.php (hunk 4 above)
Comment (iNewLegend) [reply core/common/modules/finder/categories/create.php]: Handled.

File: core/base/document.php [hunk 5]
```diff
@@ -7,11 +7,14 @@
...
 use Elementor\Controls_Manager;
 use Elementor\Controls_Stack;
+use Elementor\TemplateLibrary\Source_Local;
 use Elementor\User;
+use Elementor\Core\Documents_Manager;
 use Elementor\Core\Settings\Manager as SettingsManager;
 use Elementor\Utils;
 use Elementor\Widget_Base;
 
+
```
[the remark this answers is not in the document]
Comment (iNewLegend) [reply core/base/document.php]: Handled.

File: tests/phpunit/elementor/core/test-documents-manager.php [hunk 6]
```diff
@@ -26,4 +23,11 @@ public function test_ajax_get_document_config__set_id_build_with_elementor() {
 			get_post_meta( $post->ID, Document::BUILT_WITH_ELEMENTOR_META_KEY, true )
 		);
 	}
+
+	public function test_get_new_post_url() {
+		$new_post_url = esc_url( Documents_Manager::get_create_new_post_url() );
```
Comment (matipojo) [inline tests/phpunit/elementor/core/test-documents-manager.php:28]: Why escaping?

File: tests/phpunit/elementor/modules/library/documents/mock-library-document.php [hunk 7]
```diff
@@ -0,0 +1,10 @@
+<?php
+namespace Elementor\Tests\Phpunit\Elementor\Modules\Library\Documents;
+
+use Elementor\Modules\Library\Documents\Library_Document as Base_Library_Document;
+
+class Mock_Library_Document extends Base_Library_Document {
+	public static function get_type() {
+		return 'libraryTypeExample';
```
Comment (matipojo) [inline tests/phpunit/elementor/modules/library/documents/mock-library-document.php:8]:
```suggestion
		return 'library-type-example';
```

**Review by matipojo — CHANGES_REQUESTED** [review]

File: tests/phpunit/elementor/core/test-documents-manager.php (hunk 6 above)
Comment (iNewLegend) [reply tests/phpunit/elementor/core/test-documents-manager.php:28]: `Documents_Manager::get_create_new_post_url` return escaped URL.

Comment (iNewLegend) [reply tests/phpunit/elementor/core/test-documents-manager.php:28]: Handled.

File: tests/phpunit/elementor/modules/library/documents/mock-library-document.php (hunk 7 above)
Comment (iNewLegend) [reply tests/phpunit/elementor/modules/library/documents/mock-library-document.php]: Handled.

File: tests/phpunit/elementor/core/test-documents-manager.php [hunk 8]
```diff
@@ -26,4 +23,11 @@ public function test_ajax_get_document_config__set_id_build_with_elementor() {
 			get_post_meta( $post->ID, Document::BUILT_WITH_ELEMENTOR_META_KEY, true )
 		);
 	}
+
+	public function test_get_new_post_url() {
+		$new_post_url =  Documents_Manager::get_create_new_post_url();
```
Comment (matipojo) [inline tests/phpunit/elementor/core/test-documents-manager.php:28]: Use it from the Plugin.

File: includes/utils.php [hunk 9]
```diff
@@ -334,20 +336,9 @@ public static function get_timezone_string() {
...
-			$query_args['template_type'] = $template_type;
-		}
-
-		$new_post_url = add_query_arg( $query_args, admin_url( 'edit.php' ) );
-
-		$new_post_url = add_query_arg( '_wpnonce', wp_create_nonce( 'elementor_action_new_post' ), $new_post_url );
+		//_deprecated_function( __FUNCTION__, '3.0.8', 'Documents_Manager::get_create_new_post_url()' );
 
-		return $new_post_url;
+		return Documents_Manager::get_create_new_post_url( $post_type, $template_type );
```
Comment (matipojo) [inline includes/utils.php:341]: Use it from Plugin.

File: includes/utils.php [hunk 10]
```diff
@@ -1,6 +1,8 @@
 <?php
 namespace Elementor;
 
+use Elementor\Core\Documents_Manager;
```
Comment (matipojo) [inline includes/utils.php:4]: Remove.

**Review by matipojo — CHANGES_REQUESTED** [review]

File: tests/phpunit/elementor/core/test-documents-manager.php (hunk 8 above)
Comment (iNewLegend) [reply tests/phpunit/elementor/core/test-documents-manager.php]: Handled https://github.com/elementor/elementor/commit/23259ba9e457c9747a2ebf03ec3d5125c40ba9c7

File: includes/utils.php (hunk 9 above)
Comment (iNewLegend) [reply includes/utils.php]: Handled https://github.com/elementor/elementor/commit/23259ba9e457c9747a2ebf03ec3d5125c40ba9c7

File: includes/utils.php (hunk 10 above)
Comment (iNewLegend) [reply includes/utils.php]: Handled https://github.com/elementor/elementor/commit/23259ba9e457c9747a2ebf03ec3d5125c40ba9c7

File: includes/utils.php [hunk 11]
```diff
@@ -334,20 +334,9 @@ public static function get_timezone_string() {
...
-		];
-
-		if ( $template_type ) {
-			$query_args['template_type'] = $template_type;
-		}
-
-		$new_post_url = add_query_arg( $query_args, admin_url( 'edit.php' ) );
-
-		$new_post_url = add_query_arg( '_wpnonce', wp_create_nonce( 'elementor_action_new_post' ), $new_post_url );
+		//_deprecated_function( __FUNCTION__, '3.0.8', 'Documents_Manager::get_create_new_post_url()' );
```
Comment (matipojo) [inline includes/utils.php:337]: Use the new deprecation module.

File: core/base/document.php [hunk 12]
```diff
@@ -186,12 +188,22 @@ public static function get_class_full_name() {
 		return get_called_class();
 	}
 
+	public static function get_create_url() {
+		$base_create_url = Documents_Manager::get_create_new_post_url( Source_Local::CPT );
```
Comment (matipojo) [inline core/base/document.php:192]: Use plugin.

**Review by matipojo — CHANGES_REQUESTED** [review]

File: includes/utils.php (hunk 11 above)
Comment (iNewLegend) [reply includes/utils.php]: Handled.

File: core/base/document.php (hunk 12 above)
Comment (iNewLegend) [reply core/base/document.php]: Handled.

File: includes/utils.php [hunk 13]
```diff
@@ -334,20 +336,9 @@ public static function get_timezone_string() {
...
-		];
-
-		if ( $template_type ) {
-			$query_args['template_type'] = $template_type;
-		}
-
-		$new_post_url = add_query_arg( $query_args, admin_url( 'edit.php' ) );
-
-		$new_post_url = add_query_arg( '_wpnonce', wp_create_nonce( 'elementor_action_new_post' ), $new_post_url );
+		Module::instance()->deprecation->deprecated_function( __FUNCTION__, '3.1.0', 'Documents_Manager::get_create_new_post_url()' );
```
Comment (matipojo) [inline includes/utils.php:339]: Use the plugin!

**Review by matipojo — CHANGES_REQUESTED** [review]

File: includes/utils.php (hunk 13 above)
Comment (iNewLegend) [reply includes/utils.php]: Handled.

**Review by matipojo — DISMISSED** [review]

**Comment by iNewLegend** [comment]
Moved https://github.com/elementor/elementor/pull/13907

**Review by matipojo — DISMISSED** [review]

File: includes/utils.php [hunk 14]
```diff
@@ -334,20 +336,9 @@ public static function get_timezone_string() {
...
-		];
-
-		if ( $template_type ) {
-			$query_args['template_type'] = $template_type;
-		}
-
-		$new_post_url = add_query_arg( $query_args, admin_url( 'edit.php' ) );
-
-		$new_post_url = add_query_arg( '_wpnonce', wp_create_nonce( 'elementor_action_new_post' ), $new_post_url );
+		Plugin::$instance->modules_manager->get_modules( 'dev-tools' )->deprecation->deprecated_function( __FUNCTION__, '3.1.0', 'Plugin::$instance->documents->get_create_new_post_url()' );
```
Comment (kobizz) [inline includes/utils.php:339]:
```suggestion
		Plugin::$instance->modules_manager->get_modules( 'dev-tools' )->deprecation->deprecated_function( __FUNCTION__, '3.2.0', 'Plugin::$instance->documents->get_create_new_post_url()' );
```

**Review by kobizz — CHANGES_REQUESTED** [review]

File: includes/utils.php (hunk 14 above)
Comment (iNewLegend) [reply includes/utils.php]: Handled.

**Review by matipojo — APPROVED** [review]

**Review by kobizz — DISMISSED** [review]

**Review by matipojo — CHANGES_REQUESTED** [review]
Please resolve the conflict.

**Review by matipojo — DISMISSED** [review]

**Review by matipojo — APPROVED** [review]

**Review by matipojo — CHANGES_REQUESTED** [review]
Please pull from 3.3.0

File: includes/utils.php [hunk 15]
```diff
@@ -371,20 +371,9 @@ public static function get_timezone_string() {
...
-		];
-
-		if ( $template_type ) {
-			$query_args['template_type'] = $template_type;
-		}
-
-		$new_post_url = add_query_arg( $query_args, admin_url( 'edit.php' ) );
-
-		$new_post_url = add_query_arg( '_wpnonce', wp_create_nonce( 'elementor_action_new_post' ), $new_post_url );
+		Plugin::$instance->modules_manager->get_modules( 'dev-tools' )->deprecation->deprecated_function( __FUNCTION__, '3.2.0', 'Plugin::$instance->documents->get_create_new_post_url()' );
```
Comment (rodolphebertozzo) [inline includes/utils.php:374]:
```suggestion
		Plugin::$instance->modules_manager->get_modules( 'dev-tools' )->deprecation->deprecated_function( __FUNCTION__, '3.3.0', 'Plugin::$instance->documents->get_create_new_post_url()' );
```

**Review by rodolphebertozzo — CHANGES_REQUESTED** [review]

**Pull request merged**